.github/workflows/ci.yml

```diff
@@ -0,0 +1,170 @@
+name: CI
+
+on:
+  schedule:
+    - cron: '0 0 * * 6'
+  push:
+    branches:
+      - '*'
+  pull_request:
+
+jobs:
+  rustfmt:
+    strategy:
+      fail-fast: false
+      matrix:
+        toolchain:
+          - 1.70.0-x86_64-unknown-linux-gnu
+          - stable-x86_64-unknown-linux-gnu
+
+    name: Rustfmt (${{ matrix.toolchain }})
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: 'Setup `${{ matrix.toolchain }}`'
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ matrix.toolchain }}
+          components: rustfmt
+
+      - name: 'Override `${{ matrix.toolchain }}'
+        run: |
+          rustup override set ${{ matrix.toolchain }}
+
+      - name: cargo-fmt
+        run: |
+          cargo fmt --all -- --check
+
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        toolchain:
+          - 1.70.0-x86_64-pc-windows-msvc
+          - 1.70.0-x86_64-apple-darwin
+          - 1.70.0-x86_64-unknown-linux-gnu
+          - stable-x86_64-pc-windows-msvc
+          - stable-x86_64-apple-darwin
+          - stable-x86_64-unknown-linux-gnu
```

Comment:
we don't need to test on all of these...

Reply:
let's try it with  just `- stable-x86_64-unknown-linux-gnu` basically; let's avoid having version numbers hard-coded into any config file